### DISCUSSION TITLE
Trigger some action via command line

### DISCUSSION THREAD

**Discussion by abizeauMaxon** [debauchee/barrier#1500] [Q&A]
Hello,

First of all, thanks for this incredible tool! My question is related to this [discussion](https://github.com/debauchee/barrier/discussions/1321). I have a similar setup where Im using program to switch monitor input (Lunar on macOS & custom made app on Windows, but mControl should works too).

My question is then the following:

Instead of having to set the shortcut in Barrier server to switch K/M to computer X, and having a second shortcut to switch the monitor to computer X (by using above application & AutoHotkey like). I would like to set a single shortcut. Is there a way to do this?

1. Does barrier action could be an executable/script/batch/etc?
2. Does barrier K/M switch could be trigger via command line? (so that I can script Barrier + Monitor script in a single call)

**Comment by gibson3659** [comment]
On Linux, I just did this externally by enabling logging to a file and using swatchdog to watch the log.  Swatchdog triggers ddccontrol to change the monitor input based on if it sees "entering" or "leaving" in the log file.
